### PR TITLE
Created Header component with logo, filter and menu.

### PR DESCRIPTION
**Pull request by yuzdzhan**
No functionality is added for filtering, design only.
Mobile version is included.

### CODE REVIEW

File: src/components/Header/CloseMenuButton.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+}
+const CloseMenuButton: FC<Props> = ({ onClick }: Props) => {
+  return (
+    <button
+      onClick={onClick}
+      className="text-primary w-4 h-4 absolute top-5 right-5 z-20">
+      <svg
+        fill="none"
+        stroke="currentColor"
+        className="text-primary fill-current"
```
Comment (RedBlueKoi) [inline src/components/Header/CloseMenuButton.tsx:14]: Duplicate use of the text class. Are you sure you need to set it twice?

File: src/components/Header/Filter.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import useIsMobile from "@/custom-hooks/useIsMobile"
+import { FC, useState } from "react"
+
+const Filter: FC = () => {
+  const isMobile = useIsMobile()
```
Comment (RedBlueKoi) [inline src/components/Header/Filter.tsx:5]: It is very nice to see the use of a custom hook! However, in this case in particular I'd say it is a bit of an overkill. You can just use [a responsive modifier](https://tailwindcss.com/docs/responsive-design#targeting-a-breakpoint-range) on tailwind classes.

**Review by RedBlueKoi — COMMENTED** [review]
Overall great job! I left a couple of comments. Please let me know if you need more info on that!

File: src/components/Header/Filter.tsx (hunk 2 above)
Comment (yuzdzhan) [reply src/components/Header/Filter.tsx:5]: I have a logic for setting and removing classes in the next lines. Thats, why I desided to create and use custom hook.

File: src/components/Header/index.tsx [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import Header from "./Header"
```
[the remark this answers is not in the document]
Comment (yuzdzhan) [reply src/components/Header/index.tsx]: I have renamed the file to .ts and made it with default export.

File: src/components/Header/CartButton.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+const CartButton: FC = () => {
+  return (
+    <button className="text-primary w-8 h-8">
+      <svg
+        className="fill-primary w-8 h-8"
+        fill="none"
+        stroke="currentColor"
+        version="1.1"
+        xmlns="http://www.w3.org/2000/svg"
+        width="800px"
```
[the remark this answers is not in the document]
Comment (yuzdzhan) [reply src/components/Header/CartButton.tsx]: Removed the svg attributes.

File: src/components/Header/Filter.tsx (hunk 2 above)
Comment (yuzdzhan) [reply src/components/Header/Filter.tsx]: I figured out how to use tailwind modifiers. Thanks!

File: src/components/Header/CloseMenuButton.tsx (hunk 1 above)
Comment (yuzdzhan) [reply src/components/Header/CloseMenuButton.tsx:12]: Removed the unnecessary class.

**Review by RedBlueKoi — APPROVED** [review]
Great job! Sorry, for taking quite some time for a review. I have a lot of things on my plate and had some small issues which are resolved, so hopefully will get on track now

**Pull request merged**